src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```
@@ -1078,6 +1079,12 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.NET.DisableStandardFrameworkResolution.targets" Condition="'$(DisableStandardFrameworkResolution)' == 'true'" />
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.NET.DesignerSupport.targets" />
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.NET.GenerateAssemblyInfo.targets" Condition="'$(UsingNETSdkDefaults)' == 'true'"/>
+  <!-- Preview lang version check is temporary to unblock test writing and needs to be removed before this is merged -->
+  <Import Project="$(MSBuildThisFileDirectory)Microsoft.NET.GenerateImplicitNamespaceImports.targets"
+    Condition="'$(DisableImplicitNamespaceImports)' != 'true'
+      and '$(Language)' == 'C#'
+      and $([MSBuild]::VersionGreaterThanOrEquals('$(_TargetFrameworkVersionWithoutV)', '6.0'))
+      and '$(LangVersion)' == 'preview'"/>
```

We're not going to require the "preview" `LangVersion` for this forever.  Shouldn't the condition be something like this?

```suggestion
      and ('$(LangVersion)' == 'preview' or ($([MSBuild]::VersionGreaterThanOrEquals('$(LangVersion)', '10.0'))))"/>
```

I also wouldn't check the target framework version here.  We should restrict the default `Import` items for C# to .NET 6 and higher, but I think if you're using the appropriate C# version, you should be able to define your own `Import` items and it should work for other target frameworks.

---

Also we might want an error or a warning if you define `Import` items but it's not supported by the current language version, otherwise it will be almost impossible to figure out why they aren't working.

---

Should we remove the LangVersion check entirely? `LangVersion` isn't always a [version string](https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/LanguageVersion.cs#L373-L388) and we shouldn't be parsing compiler code in the SDK. If you remove this, I think the compiler will produce a reasonable error about it. FYI @jaredpar

---

Looking at this again, I agree that we shouldn't be in the business of resolving langversions (e.g. latest => 10.0) and making decisions based off of it. The compiler error message is pretty clear so I'll remove the lang version check.

I'll also remove the TFM check here.